sisl/io/xsf.py

```diff
@@ -378,10 +378,8 @@ class axsfSile(Sile):
     """
 
     def _setup(self, *args, **kwargs):
+        kwargs["_md_steps"] = kwargs.get("_md_steps", 1)
```

Comment:
I don't think this belongs in this PR, this PR is already quite extensive and this should go in another PR.

Reply:
It is necessary because otherwise, the later access to `_md_steps` causes it to try to open the file for reading, and then it fails the tests (where it's given a nonexistent file).

Reply:
Actually come to think of it the precise line you commented on is not necessary. Just the deleted ones below were necessary to delete. Does the `_md_steps` even matter?

Reply:
I think it should inherit from xsfSile instead of Sile and then it would matter to pass the `steps` parameter instead of `_md_steps`. But like you said that might be for another PR. Right now I guess it should just pass the tests.

Reply:
Agreed, I have fixed the inheritance in `axsf`. So if you rebase it should be fixed.
Weird I haven't had errors before?